File: priv/combined_metrics/samples/variable_naming/name_contains_type_suffix/bad/report.rb
# Report generation with type suffixes in variable names.
# BAD: variables include redundant type suffixes like _string, _list, _integer, _array, _hash.

class ReportBad
  def generate(params)
    user_string = format_user_string(params[:user])
    date_string = params[:date].strftime('%Y-%m-%d')
    title_string = build_title_string(params[:report_type])

    row_array = fetch_row_array(params[:filters])
    column_array = params[:columns]
    tag_array = params[:tags] || []

    count_integer = row_array.length
    page_count_integer = (count_integer.to_f / params[:page_size]).ceil
    total_integer = sum_total_integer(row_array)

    result_hash = build_result_hash(row_array, column_array)
    summary_hash = compute_summary_hash(row_array)

    {
      title: title_string,
      generated_by: user_string,
      generated_on: date_string,
      rows: row_array,
      tags: tag_array,
      count: count_integer,
      pages: page_count_integer,
      total: total_integer,
      result: result_hash,
      summary: summary_hash
    }
  end

  def export_report(report, format_string)
    header_array = extract_header_array(report)
    data_array = extract_data_array(report)

    case format_string
    when 'csv'
      csv_string = render_csv_string(header_array, data_array)
      { ok: true, data: csv_string }
    when 'json'
      json_string = report.to_json
      { ok: true, data: json_string }
    else
      { ok: false, error: "Unsupported format: #{format_string}" }
    end
  end

  def filter_rows(row_array, criteria_hash)
    row_array.select do |row|
      criteria_hash.all? { |key_string, value| row[key_string.to_sym] == value }
    end
  end

  def aggregate(row_array, group_by_array)
    result_hash = {}
    row_array.each do |row|
      key_string = group_by_array.map { |field| row[field] }.join('|')
      result_hash[key_string] = (result_hash[key_string] || 0) + 1
    end
    result_hash
  end

  def paginate(row_array, page_integer, page_size_integer)
    start_integer = (page_integer - 1) * page_size_integer
    row_array[start_integer, page_size_integer] || []
  end

  private

  def format_user_string(user) = "#{user[:first_name]} #{user[:last_name]}"
  def build_title_string(type) = "#{type} Report"
  def fetch_row_array(_filters) = []
  def sum_total_integer(rows) = rows.sum { |r| r[:amount] || 0 }
  def build_result_hash(rows, cols) = { rows: rows, columns: cols }
  def compute_summary_hash(rows) = { count: rows.length }
  def extract_header_array(report) = report.keys
  def extract_data_array(report) = [report.values]
  def render_csv_string(headers, data) = "#{headers.join(',')}\n#{data.inspect}"
end

File: priv/combined_metrics/samples/variable_naming/name_contains_type_suffix/bad/report.js
// Report generation with type suffixes in variable names.
// BAD: variables include redundant type suffixes like String, List, Integer, Array, Hash.

function generate(params) {
  const userString = formatUserString(params.user);
  const dateString = params.date.toISOString().slice(0, 10);
  const titleString = buildTitleString(params.reportType);

  const rowArray = fetchRowArray(params.filters);
  const columnArray = params.columns;
  const tagArray = params.tags || [];

  const countInteger = rowArray.length;
  const pageCountInteger = Math.ceil(countInteger / params.pageSize);
  const totalInteger = sumTotalInteger(rowArray);

  const resultHash = buildResultHash(rowArray, columnArray);
  const summaryHash = computeSummaryHash(rowArray);

  return {
    title: titleString,
    generatedBy: userString,
    generatedOn: dateString,
    rows: rowArray,
    tags: tagArray,
    count: countInteger,
    pages: pageCountInteger,
    total: totalInteger,
    result: resultHash,
    summary: summaryHash,
  };
}

function exportReport(report, formatString) {
  const headerArray = extractHeaderArray(report);
  const dataArray = extractDataArray(report);

  if (formatString === 'csv') {
    const csvString = renderCsvString(headerArray, dataArray);
    return { ok: true, data: csvString };
  }

  if (formatString === 'json') {
    const jsonString = JSON.stringify(report);
    return { ok: true, data: jsonString };
  }

  return { ok: false, error: `Unsupported format: ${formatString}` };
}

function filterRows(rowArray, criteriaHash) {
  return rowArray.filter(row =>
    Object.entries(criteriaHash).every(([keyString, value]) => row[keyString] === value)
  );
}

function aggregate(rowArray, groupByArray) {
  const resultHash = {};

  for (const row of rowArray) {
    const keyString = groupByArray.map(field => row[field]).join('|');
    resultHash[keyString] = (resultHash[keyString] || 0) + 1;
  }

  return resultHash;
}

function paginate(rowArray, pageInteger, pageSizeInteger) {
  const startInteger = (pageInteger - 1) * pageSizeInteger;
  const endInteger = startInteger + pageSizeInteger;
  return rowArray.slice(startInteger, endInteger);
}

function formatUserString(user) { return `${user.firstName} ${user.lastName}`; }
function buildTitleString(type) { return `${type} Report`; }
function fetchRowArray() { return []; }
function sumTotalInteger(rows) { return rows.reduce((acc, r) => acc + (r.amount || 0), 0); }
function buildResultHash(rows, cols) { return { rows, columns: cols }; }
function computeSummaryHash(rows) { return { count: rows.length }; }
function extractHeaderArray(report) { return Object.keys(report); }
function extractDataArray(report) { return [Object.values(report)]; }
function renderCsvString(headers, data) { return headers.join(',') + '\n' + JSON.stringify(data); }

module.exports = { generate, exportReport, filterRows, aggregate, paginate };
